### ISSUE TITLE
Starting from pyling 4.0.0 we get C0103 errors on naming for typing typedefs

### ISSUE THREAD

**Issue by baruchoxman** [pylint-dev/pylint#10672]
### Bug description

```python
Here's example.py


MyExampleType = typing.TypedDict(
    "MyExampleType",
    {
        "some_field": str,
    },
)
_ParamT = typing.ParamSpec("_ParamT")

Error:
example.py:22:0: C0103: Constant name "MyExampleType" doesn't conform to UPPER_CASE naming style (invalid-name)
example.py:42:0: C0103: Parameter specification variable name "_ParamT" doesn't conform to predefined naming style (invalid-name)
```

### Configuration

```ini

```

### Command used

```shell
pylint example.py
```

### Pylint output

```python
example.py:22:0: C0103: Constant name "MyExampleType" doesn't conform to UPPER_CASE naming style (invalid-name)
example.py:42:0: C0103: Parameter specification variable name "_ParamT" doesn't conform to predefined naming style (invalid-name)
```

### Expected behavior

I would not want to get these errors

### Pylint version

```shell
pylint 4.0.1
astroid 4.0.1
python 3.12.11
```

### Additional dependencies

```python

```

**Comment by jacobtylerwalls** [comment]
Thanks, can you provide the output from running pylint with `--including-naming-hint=y`? That would help us determine if the ParamSpec example is just not matching your regex.

**Comment by baruchoxman** [comment]
> --including-naming-hint=y

@jacobtylerwalls here's the output with the regex pattern:
```
example.py:42:0: C0103: Parameter specification variable name "_ParamT" doesn't conform to predefined naming style ('^_{0,2}(?:[A-Z]+|(?:[A-Z]+[a-z]+)+(?:P)?(?<!Type))$' pattern) (invalid-name)
```

**Comment by jacobtylerwalls** [comment]
Ah, then the solution is just to change `_ParamT` to `_ParamP` (or adjust the regex via the provided option)

**Comment by baruchoxman** [comment]
> option

Thanks, I realize now the convention difference, will fix the naming as suggested.
